core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -693,9 +693,10 @@ public String toString(@Nullable AbstractBlockChain chain) {
             }
             s.append(String.format(Locale.US, "%n"));
         }
-        Coin fee = getFee();
+        final Coin fee = getFee();
         if (fee != null)
-            s.append("     fee  ").append(fee.toFriendlyString()).append(String.format(Locale.US, "%n"));
+            s.append("     fee  ").append(fee.multiply(1000).divide(unsafeBitcoinSerialize().length).toFriendlyString())
```

Comment:
Actually, for a long time now, feerate has been calculated per byte. (It used to be per 1000 bytes though.)